### ISSUE TITLE
Override in azurerm_cdn_frontdoor_firewall_policy managed rules when using Microsoft_DefaultRuleSet 2.0 fails

### ISSUE THREAD

**Issue by BigFrog-coding** [hashicorp/terraform-provider-azurerm#19088]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

v1.3.3

### AzureRM Provider Version

v3.29.1

### Affected Resource(s)/Data Source(s)

azurerm_cdn_frontdoor_firewall_policy

### Terraform Configuration Files

```hcl
resource "azurerm_cdn_frontdoor_firewall_policy" "foobar" {
  name                = "foobar"
  resource_group_name = "foo"
  sku_name            = "Premium_AzureFrontDoor"
  enabled             = true
  mode                = "Detection"

  managed_rule {
    type    = "Microsoft_DefaultRuleSet"
    version = "2.0"
    action  = "Log"
    override {     
      rule_group_name = "PHP"       
      rule {
        rule_id = "933100"
        enabled = false
        action  = "Block"
      }
    }    
  }
}
```

### Debug Output/Panic Output

```shell
╷
│ Error: updating Front Door Firewall Policy: (Front Door Web Application Firewall Policy Name "foobar" / Resource Group "foo"): frontdoor.PoliciesClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="BadRequest" Message="WebApplicationFirewallPolicy validation failed. More information \"The configured rule level action is not supported for this managed rule set.\"."
│
│   with azurerm_cdn_frontdoor_firewall_policy.foobar,
│   on fd_cdn.tf line 22, in resource "azurerm_cdn_frontdoor_firewall_policy" "foobar":
│   22: resource "azurerm_cdn_frontdoor_firewall_policy" "foobar" {
│
```

### Expected Behaviour

Deploys and sets the override as expected.  I pulled this code directly from the registry https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/cdn_frontdoor_firewall_policy

### Actual Behaviour

Deployment fails when using override. I have tried few different valid overrides and they all fail with the same error.

### Steps to Reproduce

Run the code above, assuming you have a resource group
See the error

### References

This issue seems to be similar:
https://github.com/hashicorp/terraform-provider-azurerm/issues/18480

**Comment by BigFrog-coding** [comment]
Why was bug label removed? The documentation says the override can be applied and it is failing. Clearly a bug

**Comment by WodansSon** [comment]
> Why was bug label removed? The documentation says the override can be applied and it is failing. Clearly a bug

Hi @BigFrog-coding, the bot removed the the label because this did not match the criteria in the bot to be labeled as a bug. The issue here appears to be that in the documentation it is configuring `DRS 1.0` and in the repro above the version of the DRS has been updated to `2.0`. I will investigate and note my findings here, for reference you can check out the DRS 2.0 [product documentation here.](https://learn.microsoft.com/en-us/azure/web-application-firewall/afds/waf-front-door-drs?tabs=drs20#drs-20)

**Comment by WodansSon** [comment]
@BigFrog-coding, I have investigated this and this is in fact a bug. This is not in the swagger so there was no way for me to know this when I implemented this resource. It appears that the API defers the valid `version`/`action` of the `override` by the `type` value (e.g. `DefaultRuleSet` or `Microsoft_DefaultRuleSet`). In version 1.0 and 1.1 (type `DefaultRuleSet`) the accepted actions are [`Allow`, `Block`, `Log`, and `Redirect`](https://learn.microsoft.com/azure/web-application-firewall/afds/afds-overview#waf-actions), however in 2.0 (type `Microsoft_DefaultRuleSet`) the only valid action is `AnomalyScoring` due to the DRS 2.0 implementing the OWASP `anomaly score` model. That is the RCA of this issue and I think I have a fix for it and am testing it now. Thank you so much for opening this issue and I hope to have the fix in the next version of the provider. 🚀
